File: backend/app/core/roles.py
from typing import Final


ROLE_ADMIN: Final[str] = "admin"
ROLE_SALES: Final[str] = "sales"
ROLE_BUSINESS: Final[str] = "business"
ROLE_FINANCE: Final[str] = "finance"
ROLE_TECHNICIAN: Final[str] = "technician"

LEGACY_ROLE_ALIASES: Final[dict[str, str]] = {
    "tech": ROLE_TECHNICIAN,
}

PRIMARY_ROLES: Final[tuple[str, ...]] = (
    ROLE_ADMIN,
    ROLE_SALES,
    ROLE_BUSINESS,
    ROLE_FINANCE,
    ROLE_TECHNICIAN,
)


def normalize_role(role: str | None) -> str | None:
    if role is None:
        return None
    return LEGACY_ROLE_ALIASES.get(role, role)

File: frontend/src/utils/roles.ts
export type AppRole =
  | 'admin'
  | 'sales'
  | 'business'
  | 'finance'
  | 'technician'
  | 'channel_ops';

export const ROLE_LABELS: Record<AppRole, string> = {
  admin: '管理员',
  sales: '销售',
  business: '商务',
  finance: '财务',
  technician: '技术员',
  channel_ops: '渠道运营',
};

export const getRoleLabel = (role?: string | null) => {
  if (!role) {
    return '未设置角色';
  }

  return ROLE_LABELS[role as AppRole] || role;
};
